### PR TITLE
Remove the outdated monitoring package in pilot, replacing it with the istio.io/pkg/monitoring package

### PR DESCRIPTION
**Pull request by carolynhu**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/monitoring.go [hunk 1]
```diff
@@ -16,44 +16,43 @@ package v2
...
-	typeTag    = monitoring.MustCreateTag("type")
+	errTag     = monitoring.MustCreateLabel("err")
+	clusterTag = monitoring.MustCreateLabel("cluster")
+	nodeTag    = monitoring.MustCreateLabel("node")
+	typeTag    = monitoring.MustCreateLabel("type")
 
 	cdsReject = monitoring.NewGauge(
 		"pilot_xds_cds_reject",
 		"Pilot rejected CSD configs.",
-		nodeTag, errTag,
```
Comment (howardjohn) [inline pilot/pkg/proxy/envoy/v2/monitoring.go]: does this not need the WithLabels like the other?

File: go.sum [hunk 2]
```diff
@@ -767,6 +767,9 @@ istio.io/operator v0.0.0-20190813061052-3b21fa19e418/go.mod h1:WudeDwe5Tdj+6dSLi
 istio.io/pkg v0.0.0-20190515193414-9332430ad747/go.mod h1:0EkPwmR0tESYjN4Ilq1D52nTBurXaQvny3r2VY4j4tw=
 istio.io/pkg v0.0.0-20190731230704-fcbac27d69d5 h1:HcASpvj/fuuABkYH9YbsTGEOT75YHyWvvFnTe229zXs=
 istio.io/pkg v0.0.0-20190731230704-fcbac27d69d5/go.mod h1:We4ZQuCbiiNfXge2GfOshBsyDXVwzFwP8703V5DcM14=
+istio.io/pkg v0.0.0-20190737041230-fcbac27d69d5/go.mod h1:We4ZQuCbiiNfXge2GfOshBsyDXVwzFwP8703V5DcM14=
+istio.io/pkg v0.0.0-20190813222952-67b4f108fe0e h1:8RI4/zsMtrjkvyUqvy5CTGyVQPL86rERoWdm1npYU2A=
+istio.io/pkg v0.0.0-20190813222952-67b4f108fe0e/go.mod h1:We4ZQuCbiiNfXge2GfOshBsyDXVwzFwP8703V5DcM14=
```
Comment (howardjohn) [inline go.sum]: can you run mod tidy and mod vendor?

see https://github.com/istio/istio/wiki/Vendor-FAQ

File: pilot/pkg/proxy/envoy/v2/monitoring.go (hunk 1 above)
Comment (carolynhu) [reply pilot/pkg/proxy/envoy/v2/monitoring.go]: it should

File: go.sum (hunk 2 above)
Comment (carolynhu) [reply go.sum]: sure, thanks for letting me know

**Comment by carolynhu** [comment]
Need to fix test failures.

**Pull request merged**